src/software/world/robot.cpp

```diff
@@ -3,52 +3,45 @@
 #include <g3log/g3log.hpp>
 
 #include "shared/constants.h"
+#include "software/world/robot_state.h"
 
 Robot::Robot(RobotId id, const Point &position, const Vector &velocity,
              const Angle &orientation, const AngularVelocity &angular_velocity,
-             const Timestamp &timestamp, unsigned int history_duration,
+             const Timestamp &timestamp, unsigned int history_size,
              const std::set<RobotCapabilities::Capability> &capabilities)
-    : id_(id),
-      positions_(history_duration),
-      velocities_(history_duration),
-      orientations_(history_duration),
-      angularVelocities_(history_duration),
-      last_update_timestamps(history_duration),
-      capabilities_(capabilities)
+    : id_(id), states_(history_size), capabilities_(capabilities)
 {
-    addStateToRobotHistory(position, velocity, orientation, angular_velocity, timestamp);
+    if (history_size <= 0)
```

Comment:
So interesting point here. Because we are providing an `unsigned int` it's actually impossible for this value to be negative. However it can be an incorrect positive value depending how the compiler wraps values.

I think just having `unsigned int` would be enough, since once compiler errors are enabled implicit conversions can't happen to we are protected that way. Currently this code really serves no purpose and can't even be tested. Thoughts?

**Edit**
This should probably check if the size is `< 1`. What happens if there is no history? All the getter functions would segfault.